objects/src/accounts/code.rs

```diff
@@ -0,0 +1,79 @@
+use super::{AccountError, Digest};
+use assembly::{parse_module, ModuleAst};
+use crypto::merkle::MerkleTree;
+use miden_core::Program; // TODO: we should be able to import it from the assembly crate
+
+// ACCOUNT CODE
+// ================================================================================================
+
+/// Describes public interface of an account.
+///
+/// Account's public interface consists of a set of Miden VM programs each defined its own MAST.
+/// Root of each MAST commits to the underlying program. We commit to the entire account interface
+/// by building a simple Merkle tree out of all procedure MAST roots.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountCode {
+    method_tree: MerkleTree,
+    module: ModuleAst,
+    // methods: Vec<Program>, commented out because Program doesn't currently implement Eq and
+    // PartialEq. Also, there might be a better way of describing a set of programs as they
+    // might share a lot of common code blocks. In a way, we want something like a Program
+    // struct but with many entry points.
+}
```

Comment:
I think we should standardize terminology a bit here. We're using 3 different terms - `Program`, `procedure`, and `method`. Can we reduce to just `method` and `Program`? (or maybe just program, but referring to an account's methods does sound more natural). (If I've misunderstood why you're using `procedure` then this comment should perhaps be clarified a bit.)

Reply:
Agreed! Fixed!